api/submission/fetch_attempts.go

```diff
@@ -0,0 +1,42 @@
+package submission
+
+import (
+    "github.com/eriq-augustine/autograder/api/core"
+    "github.com/eriq-augustine/autograder/db"
+    "github.com/eriq-augustine/autograder/model"
+)
+
+type FetchAttemptsRequest struct {
+    core.APIRequestAssignmentContext
+    core.MinRoleGrader
+
+    TargetUser core.TargetUserSelfOrGrader `json:"target-email"`
+}
+
+type FetchAttemptsResponse struct {
+    FoundUser        bool `json:"found-user"`
+    GradingResults []*model.GradingResult `json:"grading-results"`
+}
+
+func HandleFetchAttempts(request *FetchAttemptsRequest) (*FetchAttemptsResponse, *core.APIError) {
+    response := FetchAttemptsResponse{
+        FoundUser:        false,
+        GradingResults:   make([]*model.GradingResult, 0),
+    }
+
+    if !request.TargetUser.Found {
+        return &response, nil;
+    }
+
+    response.FoundUser = true;
+
+    gradingResults, err := db.GetAttempts(request.Assignment, request.TargetUser.Email);
+    if err != nil {
+        return nil, core.NewInternalError("-607", &request.APIRequestCourseUserContext, "Failed to get submissions.").
```

Comment:
s/"submission"/"submission attempts"/

(In case this looks confusing, I am doing a sed-style find replace shorthand to tell you to replace '"submission'" with '"submission attempts"'.
I know that this explanation makes the shorthand moot, but now I'll use the shorthand whenever.)